onnx/defs/nn/defs.cc

```diff
@@ -2832,4 +2832,447 @@ ONNX_OPERATOR_SET_SCHEMA(
+                          : "NumReducedAxes = Neg (Axis1D)") // [-axis] : 1D tensor
+                  .Add(
+                      "SuffixShape = ConstantOfShape (NumReducedAxes)",
+                      "value",
+                      mktensor(1)) // [1, ..., 1] for reduced axes
+                  .Add("ReducedShape = Concat <axis = 0> (PrefixShape, SuffixShape)") // [d[0], ..., d[axis-1], 1, ..., 1]
+                  .Add("X2D = Flatten (X)", "axis", axis)
+                  .Add("XU = Cast (X2D)", "to", U);
+              builder.Add("Axes_1 = Constant()", "value", mktensor(1))
+                    .Add("Mean2D = ReduceMean (XU, Axes_1)")
+                    .Add("Exponent = Constant <value = float {2.0}>()")
+                    .Add("Square = Pow (XU, Exponent)")
+                    .Add("MeanOfSquare = ReduceMean (Square, Axes_1)");
+              builder.Add("RMS = Pow (Mean2D, Exponent)")
+                  .Add("RMSPlusEpsilon = Add (RMS, Epsilon)")
+                  .Add("StdDev = Sqrt (VarPlusEpsilon)")
+                  .Add("Normalized = Div (XU, StdDev)")
+                  .Add("NormalizedT = Cast (Normalized)", "to", T)
+                  .Add("Scale2D = Flatten <axis = 0> (Scale)")
+                  .Add("Scaled = Mul (NormalizedT, Scale2D)");
+              builder.Add("Y = Reshape (Scaled, XShape)");
+              builder.Add("InvStdDev2D = Reciprocal (StdDev)");
+              if (ctx.hasOutput(1))
+                builder.Add("Mean = Reshape (Mean2D, ReducedShape)");
+              if (ctx.hasOutput(2))
+                builder.Add("InvStdDev = Reshape (InvStdDev2D, ReducedShape)");
+
+            schema.BuildFunction(functionProto);
+            return true;
+          }));
+
+static const char* SkipLayerNormalization_ver23_doc = R"DOC(
+
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    SkipLayerNormalization,
+    23,
+    OpSchema()
+        .SetDoc(SkipLayerNormalization_ver23_doc)
+        .Attr("epsilon",
+              "The epsilon value to use to avoid division by zero.",
+              AttributeProto::FLOAT, 1e-5f)
+        .Input(0,
+               "X",
+               "3D input tensor with shape (batch_size, sequence_length, hidden_size)"
+               "Or 2D input tensor with shape (token_count, hidden_size)",
+               "T")
+        .Input(1,
+               "S",
+               "3D input tensor with shape (batch_size, sequence_length, hidden_size)"
+               "Or 2D input tensor with shape (token_count, hidden_size)",
+               "T")
+        .Input(2,
+               "gamma",
+               "1D input tensor with shape (hidden_size)",
+               "T")
+        .Input(3,
+               "beta",
+               "1D skip tensor with shape (hidden_size)",
+               "T",
+               OpSchema::Optional)
+        .Input(4,
+               "B",
+               "1D bias tensor with shape (hidden_size)",
```

Comment:
So, is this different from a layer-norm's bias? In LayerNorm, the bias seems to be added to the output of normalization. Here, it looks like it is added to the input. Is that intentional?